lib/include/ert/ecl/ecl_sum_file_data.hpp

```diff
@@ -0,0 +1,67 @@
+
+#include <ert/util/vector.hpp>
+#include <ert/util/double_vector.h>
+
+#include <ert/ecl/ecl_smspec.hpp>
+#include <ert/ecl/ecl_sum_tstep.hpp>
+#include <ert/ecl/ecl_file.hpp>
+
+namespace ecl {
+
+#define INVALID_MINISTEP_NR -1
+#define INVALID_TIME_T 0
+
+class ecl_sum_file_data_type {
+
+  double  days_start;
+  double  sim_length;
+  int first_report_step;
+  int last_report_step;
+  ecl_smspec_type * smspec;
+  vector_type * data;
+  int_vector_type * report_first_index;
+  int_vector_type * report_last_index;
+  bool         index_valid;
+  time_t       start_time;             /* In the case of restarts the start might disagree with the value reported
+                                          in the smspec file. */
+  time_t       end_time;
+
+
+  void                 clear_index();
+  void                 append_tstep__(ecl_sum_tstep_type * tstep);
+
+  void                 build_index();
+  bool                 has_report_step(int report_step ) const;
+  void                 report2internal_range(int report_step , int * index1 , int * index2 ) const;
+  void                 fwrite_report__( int report_step , fortio_type * fortio) const;
+  bool                 check_file( ecl_file_type * ecl_file );
+  void                 add_ecl_file(int report_step, const ecl_file_view_type * summary_view, const ecl_smspec_type * smspec);
+
+  public:
```

Comment:
Can you please reorder the content of the class so the public stuff comes first, and the private at the bottom.